lib/semian/net_http.rb

```diff
@@ -43,10 +43,24 @@ def semian_identifier
       ::SystemCallError, # includes ::Errno::EINVAL, ::Errno::ECONNRESET, ::Errno::ECONNREFUSED, ::Errno::ETIMEDOUT, and more
     ].freeze # Net::HTTP can throw many different errors, this tries to capture most of them
 
+    module ClassMethods 
+      def new(*args, semian: true)
+        http = super(*args)
+        http.instance_variable_set(:@semian_enabled, semian)
+        http
+      end
+    end
+
     class << self
       attr_accessor :exceptions
       attr_reader :semian_configuration
 
+      def prepended(base)
+        class << base
+          prepend ClassMethods
+        end
+      end
```

Comment:
Not a fan of this, you can just do `Net::HTTP.singleton_class.prepend(Semian::NetHTTP::ClassMethods)`